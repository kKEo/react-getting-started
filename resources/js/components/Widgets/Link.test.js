import React from 'react';
import Link from './Link';
import renderer from 'react-test-renderer';

it ('renders correctly', () => {
    const tree = renderer
        .create(<Link page="https://centrumrekodziela.pl">Centrum</Link>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders as an anchor when no page is set', () => {
    const tree = renderer.create(<Link>Facebook</Link>).toJSON();
    expect(tree).toMatchSnapshot();
});

it('properly escapes quotes', () => {
   const tree = renderer
       .create(<Link>{"\"Everythink\" \\'is\\ 'awesome'"}</Link>)
       .toJSON();
   expect(tree).toMatchSnapshot();
});
