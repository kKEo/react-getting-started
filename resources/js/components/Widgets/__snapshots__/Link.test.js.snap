// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`properly escapes quotes 1`] = `
<a
  className="normal"
  href="#"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  "Everythink" \\'is\\ 'awesome'
</a>
`;

exports[`renders as an anchor when no page is set 1`] = `
<a
  className="normal"
  href="#"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Facebook
</a>
`;

exports[`renders correctly 1`] = `
<a
  className="normal"
  href="https://centrumrekodziela.pl"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Centrum
</a>
`;
